### PR TITLE
Add MathML Support to `@types/react`

### PR DESCRIPTION
**Pull request by louisscruz**
This adds support to React for [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) as [discussed previously](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69884).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/MathML
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

## Notes

In initial discussion, it was brought up that:

> If it works at runtime, we should support it with types.

MathML is still [newly supported by modern browsers](https://caniuse.com/mathml). As a result, the browser support for some elements – and especially for some element attributes – is mixed. In some cases, elements have been deprecated and/or have had support removed. In other cases, the support across browsers is "non-standard", as MDN puts it. For example, various alignment and styling attributes on the `mtable` element are not supported in Chrome (v126 at the time of this writing), but are supported in Firefox. This can be seen by [previewing the example from MDN](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mtable#alignment_with_row_number):

```html
<mtable frame="solid" rowlines="solid" align="axis 3">
```

(Chrome): 
Screenshot 2024-07-06 at 3 21 09 PM

(Firefox): 
Screenshot 2024-07-06 at 3 21 30 PM

~~In regards to the spectrum of support, the approach initially taken in this PR is as follows:~~

* ~~All elements MDN lists, except for those marked as deprecated and/or non-standard, have relevant types added in this PR.~~
* ~~All attributes MDN lists for a given element, except for those marked as deprecated, have relevant types added in this PR. Note that types for  "non-standard" attributes have been added in this PR.~~

~~This approach was taken to reduce the initial blast radius of the initial MathML support with the intent that more detailed elements/attributes with further afield support could be added if the community finds a need to reach for them. That being said, I'm happy to discuss this more and adjust to whatever the maintainers of these types see as the best approach.~~

The difference in seems to come from the difference between MathML and MathML Core from the Math WG:

* [MathML 3 specification](https://www.w3.org/TR/MathML3/)
* [MathML Core specification](https://www.w3.org/TR/mathml-core/)

As MDN puts summarizes:

> MDN uses [MathML Core](https://w3c.github.io/mathml-core/) as a reference specification but, due to an erratic standardization history, legacy MathML features may still show up in existing implementations and web content.

and

> Note: It is highly recommended that developers and authors switch to MathML Core, perhaps relying on other web technologies to cover missing use cases. The Math WG is maintaining a set of [MathML polyfills](https://github.com/mathml-refresh/mathml-polyfills) to facilitate that transition.

The approach taken in this PR is to use the MathML Core specification. **Note: Updates still need to be made to fully support this specification**.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d960a9fbc6b8e027e7e33aaa06694b8db9bd9914
</p>

File: types/react/index.d.ts [hunk 1]
```diff
@@ -3922,6 +3924,145 @@ declare namespace React {
...
+    }
+    interface MathMLMunderAttributes extends MathMLAttributes<MathMLMunderElement> {
+        accentunder?: boolean | undefined;
+    }
+    interface MathMLMunderoverAttributes extends MathMLAttributes<MathMLElement> {
+        accent?: boolean | undefined;
+        accentunder?: boolean | undefined;
+    }
+    /**
+     * @see https://w3c.github.io/mathml-core/#semantics-and-presentation
```
Comment (louisscruz) [inline types/react/index.d.ts:4050]: This and the urls below should be `link`s like used elsewhere for better in-editor readability.

**Comment by eps1lon** [comment]
Is there a spec for MathML? For HTML and SVG we follow the spec instead of what browsers implement and let users augment their types with non-standard elements/attributes.

**Comment by louisscruz** [comment]
That makes sense. Yes, there are specifications. I added some more context on that in the PR description.

It ends up being that the Math WG is quite a bit older than I'd previously realized.... apparently, the group has been in operation for ~26 years! MDN describes the MathML standardization history as "erratic", and the Math WG describes MathML Core as "a subset that can be reliably displayed in web browsers". It seems like MathML Core is perhaps the thing to support instead of MathML (v3). [That seems to also be the route Chrome has taken](https://chromestatus.com/feature/5240822173794304).

**Comment by louisscruz** [comment]
For now, I'll move forward with adjusting the PR to support the MathML Core specification. I'll take a pass in the next few days.

**Comment by louisscruz** [comment]
Sorry for delay here, planning to take a look this week (and push up some changes).

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please address feedback

**Comment by louisscruz** [comment]
Ah, I'd forgotten about this PR! I'm planning to continue this proposed addition over in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71187

**Pull request closed without merging**